src/Umbraco.Core/UriExtensions.cs

```diff
@@ -137,6 +138,14 @@ internal static bool IsDefaultBackOfficeRequest(this Uri url, IGlobalSettings gl
             return false;
         }
 
+        /// <summary>
+        /// Non Client Side request Extensions <see cref="IsClientSideRequest"/>
+        /// </summary>
+        internal readonly static IDictionary<string, bool> NonClientSideRequestExtensions = new Dictionary<string, bool>(5,StringComparer.InvariantCultureIgnoreCase)
```

Comment:
@nzdev feel free to correct me if I'm wrong, but wouldn't a `HashSet<string>` be even better here? Since all values in the dictionary are `true`, it seems like we only really need the key.

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=netframework-4.7.2#System_Collections_Generic_Dictionary_2__ctor_System_Int32_System_Collections_Generic_IEqualityComparer__0__

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.-ctor?view=netframework-4.7.2#System_Collections_Generic_HashSet_1__ctor_System_Int32_System_Collections_Generic_IEqualityComparer__0__

Reply:
Makes sense to me

Reply:
Thanks for the feedback @abjerner - I've applied that change now.